src/plugins/inspector/public/views/requests/components/disambiguate_request_names.ts

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { groupBy } from 'lodash';
+import type { Request } from '../../../../common/adapters/request/types';
+
+export function disambiguateRequestNames(requests: Request[]): Request[] {
+  const requestsByName = groupBy(requests, (r) => r.name);
+
+  const newNamesById = Object.entries(requestsByName).reduce<{ [requestId: string]: string }>(
+    (acc, [name, reqs]) => {
+      const next = { ...acc };
```

Comment:
isn't creating new object here redundant?

Reply:
Yea, it is not necessary, kind of did it as a reflex - will remove!